getUserMedia.js

```diff
@@ -38,39 +30,31 @@ export default function getUserMedia(
         } else if (typeofMediaTypeConstraints == 'object') {
           ++requestedMediaTypes;
         } else {
-          errorCallback(
-            new TypeError(
-              'constraints.' + mediaType
-                + ' is neither a boolean nor a dictionary'));
-          return;
+          throw new TypeError('constraints.' + mediaType + ' is neither a boolean nor a dictionary');
         }
       }
     }
     // According to step 3 of the getUserMedia() algorithm, if
     // requestedMediaTypes is the empty set, the method invocation fails with
     // a TypeError.
     if (requestedMediaTypes === 0) {
-      errorCallback(new TypeError('constraints requests no media types'));
-      return;
+      throw new TypeError('constraints requests no media types');
     }
   } else {
-    errorCallback(new TypeError('constraints is not a dictionary'));
-    return;
+    throw new TypeError('constraints is not a dictionary');
   }
 
-  WebRTCModule.getUserMedia(
-    constraints,
-    /* successCallback */ (id, tracks) => {
+  return WebRTCModule.getUserMedia(constraints)
+    .then(([id, tracks]) => {
       const stream = new MediaStream(id);
       for (const track of tracks) {
         stream.addTrack(new MediaStreamTrack(track));
       }
-
-      successCallback(stream);
-    },
-    /* errorCallback */ (type, message) => {
+      return stream;
+    })
+    .catch(({ message, code }) => {
```

Comment:
shouldn't it be `{ code, message }` ?

Reply:
That shouldn't matter:

```javascript
const obj = {a: 1, b: 2};
const f1 = ({a, b}) => console.log(a);
const f2 = ({b, a}) => console.log(a);
f1(obj); // => 1
f2(obj); // => 1
```

Reply:
@chroth7 

your right, my eyes was ignoring `{}` mysteriously
thanks